Everything works with python 3.10 except the optional dependency job. Looks like the optional dependencies fail on Python 3.10 in the llvmjit code. There's no useful output from the test runner (it seems to just crash). I think I've seen that error before from Travis before most travis jobs were disabled. Probably the temporary fix there is to disable whatever tests are crashing the test runner so we can at least see that the other tests pass.

For 3.11b1 the optional dependency test fails because there are no wheels for numpy and building numpy 1.22.3 from source fails with a compiler error.

The 3.11b1 doctest and test2 jobs fail because of some problem with pickling. The change comes from https://github.com/python/cpython/issues/70766 and gives errors like
```python
___________________ sympy.polys.polyutils.PicklableWithSlots ___________________
File "/home/runner/work/sympy/sympy/sympy/polys/polyutils.py", line 455, in sympy.polys.polyutils.PicklableWithSlots
Failed example:
    some2 = loads(dumps(some))
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.11.0-beta.1/x64/lib/python3.11/doctest.py", line 1346, in __run
        exec(compile(example.source, filename, "single",
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "<doctest sympy.polys.polyutils.PicklableWithSlots[9]>", line 1, in <module>
        some2 = loads(dumps(some))
                      ^^^^^^^^^^^
      File "/home/runner/work/sympy/sympy/sympy/polys/polyutils.py", line 474, in __getstate__
        d.update(c.__getstate__(self, c))
                 ^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/runner/work/sympy/sympy/sympy/polys/polyutils.py", line 474, in __getstate__
        d.update(c.__getstate__(self, c))
                 ^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: object.__getstate__() takes no arguments (1 given)
```
I've just pushed a quick fix for that here.

--- a/.github/workflows/runtests.yml
+++ b/.github/workflows/runtests.yml
@@ -255,11 +255,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', 'pypy-3.8']
+        python-version: ['3.8', '3.9', '3.11.0-alpha - 3.11', 'pypy-3.8']
         experimental: [false]
-        include:
-          - python-version: '3.11.0-alpha - 3.11'
-            experimental: true
+        # Maybe use this to add 3.12 when the time comes:
+        #include:
+        #  - python-version: '3.11.0-alpha - 3.11'
+        #    experimental: true
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v3
@@ -304,11 +305,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', 'pypy-3.8']
+        python-version: ['3.8', '3.9', '3.11.0-alpha - 3.11', 'pypy-3.8']
         experimental: [false]
-        include:
-          - python-version: '3.11.0-alpha - 3.11'
-            experimental: true
+        # Maybe use this to add 3.12 when the time comes:
+        #include:
+        #  - python-version: '3.11.0-alpha - 3.11'
+        #    experimental: true
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v3

--- a/sympy/polys/polyutils.py
+++ b/sympy/polys/polyutils.py
@@ -470,8 +470,15 @@ class PicklableWithSlots:
 
         # Get all data that should be stored from super classes
         for c in cls.__bases__:
-            if hasattr(c, "__getstate__"):
-                d.update(c.__getstate__(self, c))
+            # XXX: Python 3.11 defines object.__getstate__ and it does not
+            # accept any arguments so we need to make sure not to call it with
+            # an argument here. To be compatible with Python < 3.11 we need to
+            # be careful not to assume that c or object has a __getstate__
+            # method though.
+            getstate = getattr(c, "__getstate__", None)
+            objstate = getattr(object, "__getstate__", None)
+            if getstate is not None and getstate is not objstate:
+                d.update(getstate(self, c))
 
         # Get all information that should be stored from cls and return the dict
         for name in cls.__slots__:

